File: atcoder.jp/PyPy3 (7.3.0)/abc116/abc116_b.py
s = int(input())

def f(n):
    if n % 2:
        return 3 * n + 1
    else:
        return n / 2

a = [s]
memo = set()
memo.add(s)

i = 1
while True:
    an = f(a[i-1])
    if an in memo:
        break
    a.append(an)
    memo.add(an)
    i += 1

print(i+1)

File: atcoder.jp/Rust (rustc 1.70.0)/abc116/abc116_b.rs
use proconio::{input, fastout};
use std::collections::HashSet;

fn f(n: usize) -> usize {
    if n % 2 == 0 {
        n / 2
    } else {
        3 * n + 1
    }
}

#[fastout]
fn main() {
    input! {
        mut s: usize,
    }

    let mut set = HashSet::new();
    set.insert(s);

    for i in 2.. {
        s = f(s);
        if set.contains(&s) {
            println!("{}", i);
            return;
        }
        set.insert(s);
    }
}
